spec/lib/api/v3/work_packages/work_package_collection_representer_spec.rb

```diff
@@ -534,13 +532,58 @@
   context 'when passing schemas' do
     let(:embed_schemas) { true }
 
-    it 'embeds a schema collection' do
+    it 'embeds a schema collection', :aggregate_failures do
       expected_path = api_v3_paths.work_package_schema(work_packages[0].project.id,
                                                        work_packages[0].type.id)
 
       expect(collection)
         .to be_json_eql(expected_path.to_json)
         .at_path('_embedded/schemas/_embedded/elements/0/_links/self/href')
+
+      # All work packages share the same project and type
+      expect(collection)
+        .to have_json_size(1)
+              .at_path('_embedded/schemas/_embedded/elements')
+    end
+
+    context 'when having timestamps leads to a different schema at the baseline time (because of a different type)',
```

Comment:
To avoid unnecessary context switching in this spec (and to just be able to grok through it a little better), how about defining

```ruby
let(:first_wp) { WorkPackage.first } # or work_packages[0]
```

and refer to `first_wp` throughout the entire context?